sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidSqlValidator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.sql.calcite.planner;
+
+import org.apache.calcite.adapter.java.JavaTypeFactory;
+import org.apache.calcite.prepare.Prepare.CatalogReader;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.sql.SqlInsert;
+import org.apache.calcite.sql.SqlOperatorTable;
+import org.apache.calcite.sql.validate.SqlConformance;
+import org.apache.calcite.sql.validate.SqlValidatorImpl;
+
+/**
+ * Clone of Calcite's {@code CalciteSqlValidator} which is not
+ * visible to Druid.
+ */
+class DruidSqlValidator extends SqlValidatorImpl
```

Comment:
hmm, should this be instead modifications to https://github.com/apache/druid/blob/master/sql/src/main/java/org/apache/calcite/prepare/DruidSqlValidator.java or does the `DruidPlanner` really need to clone the validator differently than the other one used by `CalcitePlanner`?

Reply:
Good catch. So, what's *supposed* to happen is that the now-renamed `BaseDruidSqlValidator` simply exposes the Java-specific `CalciteSqlValidator` by living in the Calcite name space and extending a protected class. Then, `DruidSqlValidator` is the Druid version, with soon-to-be-added extensions for the `INSERT` statement. The latest commit makes this clearer: `DruidSqlValidator` extends `BaseDruidSqlValidator` and removes the methods we now inherit from `CalciteSqlValidator`. Clear as mud?

Let's let Travis do its thing to validate this latest adjustment, then please take another look.